massa-execution-worker/src/active_history.rs

```diff
@@ -0,0 +1,120 @@
+use massa_execution_exports::ExecutionOutput;
+use massa_hash::Hash;
+use massa_ledger_exports::{
+    LedgerEntry, LedgerEntryUpdate, SetOrDelete, SetOrKeep, SetUpdateOrDelete,
+};
+use massa_models::{Address, Amount};
+use std::collections::VecDeque;
+
+#[derive(Default)]
+/// History of the outputs of recently executed slots.
+/// Slots should be consecutive, oldest at the beginning and latest at the back.
+pub(crate) struct ActiveHistory(pub VecDeque<ExecutionOutput>);
+
+/// Result of a lazy, active history search
+pub enum HistorySearchResult<T> {
+    Present(T),
+    Absent,
+    NoInfo,
+}
+
+impl ActiveHistory {
+    /// Lazily query (from end to beginning) the active balance of an address after a given index.
+    ///
+    /// Returns a `HistorySearchResult`.
+    pub fn fetch_active_history_balance(
+        &self,
+        addr: &Address,
+        index: Option<usize>,
+    ) -> HistorySearchResult<Amount> {
+        let iter = self
+            .0
+            .iter()
+            .take(index.unwrap_or_default().saturating_add(1))
```

Comment:
uuh I think that we should not use unwrap_or_default but:
`.take(index.map_or_else(|| history_length, |v| v.saturating_add(1) )`

so that when the index is not specified, we get the whole history instead of nothing